static_handler.cc

```diff
@@ -24,29 +30,200 @@ const std::unordered_map<std::string,std::string> content_mappings
 
 RequestHandler::Status StaticHandler::Init(const std::string& uri_prefix, const NginxConfig& config)
 {
-	// initialiing uri prefix and directory
+	// initializing variables
 	prefix_ = uri_prefix;
+	timeout_ = -1;
 
 	for (auto statement : config.statements_)
 	{
+		//look for root
 		if(statement->tokens_.size() == 2 && statement->tokens_[0] == "root")
 		{
 			directory_ = statement->tokens_[1];
-			return RequestHandler::OK;
+		}
+
+		//look for authetication list for private files (optional)
+		else if(statement->tokens_.size() == 2 && statement->tokens_[0] == "authentication_list")
+		{
+			if(!init_authentication_database(statement->tokens_[1]))
+			{
+				BOOST_LOG_TRIVIAL(error) << "error parsing given authentication_list file";
+				return RequestHandler::ERROR;
+			}
+		}
+
+		//look for timeout value in seconds for authentication (optional)
+		else if(statement->tokens_.size() == 2 && statement->tokens_[0] == "timeout")
+		{
+			//exception will be thrown and caught from main if value isnt an int
+			timeout_ = std::stoi(statement->tokens_[1]);
 		}
 	}
 
-	BOOST_LOG_TRIVIAL(error) << "root not specified in static handler for " << uri_prefix;
-	return RequestHandler::ERROR;
+	//root not set
+	if(directory_ == "")
+	{
+		BOOST_LOG_TRIVIAL(error) << "root not specified in static handler for " << uri_prefix;
+		return RequestHandler::ERROR;
+	}
+
+	return RequestHandler::OK;
 }
 
-RequestHandler::Status StaticHandler::HandleRequest(const Request& request, Response* response) const
+RequestHandler::Status StaticHandler::HandleRequest(const Request& request, Response* response)
 {
 	BOOST_LOG_TRIVIAL(trace) << "Creating static file response";
 
 	std::string full_path = request.uri();
 	std::string file_path = directory_ + full_path.substr(prefix_.length()); // get file path relative to server
 
+	// Check if cookie exists if authentication
```

Comment:
Good idea, we'll keep it as is for this pull request though and will try to refactor later if we have time.